[CI] DISIAccumulatorTests testAllDocs failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12300 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/g562naqgdvcqq)
- [elasticsearch-periodic-platform-support #12300 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/pkhoujqd2htgi)

**Reproduction Line:**
```
null
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.tsdb.es819.DISIAccumulatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testAllDocs'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: startIndex=1, numBits=1
```

**Issue Reasons:**
- [9.3] 1 consecutive failures in test testAllDocs
- [9.3] 2 consecutive failures in step debian-12_platform-support-unix
- [9.3] 1 consecutive failures in pipeline elasticsearch-periodic-platform-support
- [9.3] 2 failures in test testAllDocs (4.8% fail rate in 42 executions)
- [9.3] 2 failures in step debian-12_platform-support-unix (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Test suite doesn't extend from `ESTestCase`, and there for the reproduction line doesn't work.

When updating `DISIAccumulatorTests` to extend from `ESTestCase`. The following failure:

```
REPRODUCE WITH: ./gradlew ":server:test" --tests "org.elasticsearch.index.codec.tsdb.DISIAccumulatorTests.testAllDocs" -Dtests.seed=DECF9345789B9873 -Dtests.locale=ca-FR -Dtests.timezone=America/St_Barthelemy -Druntime.java=26

startIndex=1, numBits=1
java.lang.AssertionError: startIndex=1, numBits=1
	at __randomizedtesting.SeedInfo.seed([DECF9345789B9873:D7CDDE2D4ECE5AD]:0)
	at org.apache.lucene.util.FixedBitSet.set(FixedBitSet.java:713)
	at org.elasticsearch.index.codec.tsdb.DISIAccumulatorTests.testAllDocs(DISIAccumulatorTests.java:212)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
```

Looks like fixed bitset is sometimes initialized with to small capacity.

--- a/server/src/test/java/org/elasticsearch/index/codec/tsdb/DISIAccumulatorTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/tsdb/DISIAccumulatorTests.java
@@ -14,13 +14,13 @@ import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.IOContext;
 import org.apache.lucene.store.IndexInput;
 import org.apache.lucene.store.IndexOutput;
-import org.apache.lucene.tests.util.LuceneTestCase;
 import org.apache.lucene.tests.util.TestUtil;
 import org.apache.lucene.util.BitSet;
 import org.apache.lucene.util.BitSetIterator;
 import org.apache.lucene.util.FixedBitSet;
 import org.apache.lucene.util.SparseFixedBitSet;
 import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
 import java.util.Random;
@@ -28,7 +28,7 @@ import java.util.Random;
 // Copied from org.apache.lucene.codecs.lucene90.TestIndexedDISI and kept tests that we can run.
 // The test suite has been modified to write jump table using writeJumpTable(...) in this class.
 // (some original tests require access to package protected constructor of IndexedDISI and was removed)
-public class DISIAccumulatorTests extends LuceneTestCase {
+public class DISIAccumulatorTests extends ESTestCase {
 
     public void testEmpty() throws IOException {
         int maxDoc = TestUtil.nextInt(random(), 1, 100000);
@@ -208,7 +208,7 @@ public class DISIAccumulatorTests extends LuceneTestCase {
     public void testAllDocs() throws IOException {
         int maxDoc = TestUtil.nextInt(random(), 1, 100000);
         FixedBitSet set = new FixedBitSet(maxDoc);
-        set.set(1, maxDoc);
+        set.set(0, maxDoc);
         try (Directory dir = newDirectory()) {
             doTest(set, dir);
         }

